File: apps/s-2-admin/src/api/availability/AvailabilityCreateInput.ts
import { UserWhereUniqueInput } from "../user/UserWhereUniqueInput";
import { EventTypeWhereUniqueInput } from "../eventType/EventTypeWhereUniqueInput";
import { ScheduleWhereUniqueInput } from "../schedule/ScheduleWhereUniqueInput";

export type AvailabilityCreateInput = {
  user?: UserWhereUniqueInput | null;
  eventType?: EventTypeWhereUniqueInput | null;
  days: number;
  startTime: Date;
  endTime: Date;
  date?: Date | null;
  schedule?: ScheduleWhereUniqueInput | null;
};

File: apps/cactus-demo-dotnet/src/APIs/Availability/Dtos/AvailabilityCreateInput.cs
namespace CactusDemoDotnet.APIs.Dtos;

public class AvailabilityCreateInput
{
    public int? Id { get; set; }

    public User? User { get; set; }

    public EventType? EventType { get; set; }

    public int Days { get; set; }

    public DateTime StartTime { get; set; }

    public DateTime EndTime { get; set; }

    public DateTime? Date { get; set; }

    public Schedule? Schedule { get; set; }
}
